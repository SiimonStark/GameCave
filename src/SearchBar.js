import React, {Component} from 'react';

export default class SearchBar extends Component {
  constructor(props) {
    super(props)
    this.state = {}
  }
  updateText = event => {
    let searchText = event.target.value.toLowerCase()
    this.props.updateState({titleFilter: searchText})
  }
  render() {
    return(
      <div className="search-bar search-section">
        <input maxLength="25" placeholder="SEARCH..." className="search" onChange={this.updateText}/>
      </div>
    )
  }
}