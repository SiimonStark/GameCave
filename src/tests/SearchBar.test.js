import React from 'react';
import ReactDOM from 'react-dom';
import SearchBar from '../SearchBar';
import { shallow } from 'enzyme';

const updateStateMock = jest.fn();

describe('SearchBar', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <SearchBar 
        updateState = {updateStateMock}/>
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call updateState on app on input change', () => {
    wrapper.find('.search').simulate('change', {target: {value: 'pam'}});
    expect(updateStateMock).toBeCalled();
  })
})