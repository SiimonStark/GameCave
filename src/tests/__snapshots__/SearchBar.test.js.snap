// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match snapshot 1`] = `
<div
  className="search-bar search-section"
>
  <input
    className="search"
    maxLength="25"
    onChange={[Function]}
    placeholder="Search a game title!"
  />
</div>
`;
